src/lib_json/CMakeLists.txt

```diff
@@ -151,7 +151,11 @@ if(BUILD_SHARED_LIBS)
 endif()
 
 if(BUILD_STATIC_LIBS)
-    set(STATIC_LIB ${PROJECT_NAME}_static)
+    if (BUILD_SHARED_LIBS)
+        set(STATIC_LIB ${PROJECT_NAME}_static)
```

Comment:
You can `add_library(<default name> ALIAS jsoncpp_static)` or `add_library(<default name> INTERFACE)` and them `target_link_libraries(<default name> INTERFACE jsoncpp_static)